pkg/controller/util/upgrade.go

```diff
@@ -0,0 +1,76 @@
+package util
+
+import (
+	"fmt"
+	"github.com/Masterminds/semver"
+	"github.com/go-logr/logr"
+	"github.com/tidwall/gjson"
+	"io/ioutil"
+	"net/http"
+	"strings"
+)
+
+const (
+	ProxyImageName = "proxy"
+
+	CollectorImageName = "wavefront-kubernetes-collector"
+
+	ImagePrefix = "wavefronthq/"
+)
+
+func GetLatestVersion(crImageName string, currentVersion string, enableAutoUpgrade bool, reqLogger logr.Logger) (string, error) {
+	if !enableAutoUpgrade {
+		return currentVersion, nil
+	}
+
+	// "latest" effectively renders auto upgrade useless.
+	if currentVersion == "latest" {
+		return currentVersion, nil
+	}
+
+	// The last 20 tags should be good. Don't expect customers to be using a really old version of CR.
+	url := "https://registry.hub.docker.com/v2/repositories/wavefronthq/" + crImageName + "/tags/?page_size=20"
```

Comment:
Should we add some checks against the image name / location before doing this? Account for openshift / other custom locations?